File: VB/tagsList/XtraForm1.vb
' Developer Express Code Central Example:
' How to create a control that shows a list of tags
' 
' This example demonstrates how to create a TagList control editor that supports
' both standalone and in-place modes.
' 
' You can find sample updates and versions for different programming languages here:
' http://www.devexpress.com/example=E4542


Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Grid

Namespace tagsList
	Partial Public Class XtraForm1
		Inherits DevExpress.XtraEditors.XtraForm
		Public Sub New()
			InitializeComponent()
			Dim t As String = "Haskell,Ocaml,Scala,Ruby"
			taglist1.EditValue = t
			Dim ds As New BindingList(Of myDataSource)()
			For i As Integer = 0 To 9
				If i < 5 Then
					ds.Add(New myDataSource() With {.Langs = "Perl,Python"})
				Else
					ds.Add(New myDataSource() With {.Langs = "Perl,Python,Java"})
				End If
			Next i
			gridControl1.DataSource = ds
			TryCast(gridControl1.MainView, GridView).Columns("Langs").ColumnEdit = New TagListReposioryItem()
		End Sub

		Protected Overrides Sub OnLoad(ByVal e As EventArgs)
			MyBase.OnLoad(e)
		End Sub
	End Class
	Public Class myDataSource
		Private _Langs As String

		Public Property Langs() As String
			Get
				Return _Langs
			End Get
			Set(ByVal value As String)
				_Langs = value
			End Set
		End Property
	End Class
End Namespace

File: CS/tagsList/XtraForm1.cs
// Developer Express Code Central Example:
// How to create a control that shows a list of tags
// 
// This example demonstrates how to create a TagList control editor that supports
// both standalone and in-place modes.
// 
// You can find sample updates and versions for different programming languages here:
// http://www.devexpress.com/example=E4542

using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Views.Grid;

namespace tagsList {
    public partial class XtraForm1 : DevExpress.XtraEditors.XtraForm {
        public XtraForm1() {
            InitializeComponent();
            string t = "Haskell,Ocaml,Scala,Ruby";
            taglist1.EditValue = t;
            BindingList<myDataSource> ds = new BindingList<myDataSource>();
            for(int i = 0; i < 10; i++) {
                if(i < 5) ds.Add(new myDataSource() { Langs = "Perl,Python" });
                else ds.Add(new myDataSource() { Langs = "Perl,Python,Java" });
            }
            gridControl1.DataSource = ds;
            (gridControl1.MainView as GridView).Columns["Langs"].ColumnEdit = new TagListReposioryItem();
        }

        protected override void OnLoad(EventArgs e) {
            base.OnLoad(e);
        }
    }
    public class myDataSource {
        private string _Langs;

        public string Langs {
            get { return _Langs; }
            set { _Langs = value; }
        }    
    }
}